src/ledger/database/hashmap.zig

```diff
@@ -20,31 +21,36 @@ pub fn SharedHashMapDB(comptime column_families: []const ColumnFamily) type {
         maps: []SharedHashMap,
         /// shared lock is required to call locking map methods.
         /// exclusive lock is required to call non-locking map methods.
-        transaction_lock: DefaultRwLock = .{},
+        /// to avoid deadlocks, always acquire the shared lock *before* acquiring the map lock.
+        transaction_lock: *RwLock,
```

Comment:
Would be good to understand why we're allocating this field now, any comment?

Reply:
The database is copied sometimes based on the assumption that it only contains pointers. If this lock is copied then it doesn't work. It needs to always use the same lock.